lib/cylc/cycling/iso8601.py

```diff
@@ -366,11 +373,13 @@ def set_offset(self, offset):
         self._cached_next_point_values = {}
         self._cached_valid_point_booleans = {}
         self._cached_recent_valid_points = []
-        self.value = str(self.recurrence)
+        self.value = str(self.recurrence) + '!' + str(self.exclusion)
 
     def is_on_sequence(self, point):
         """Return True if point is on-sequence."""
         # Iterate starting at recent valid points, for speed.
+        if point == self.exclusion:
```

Comment:
You weren't tempted to go for the same `if exclusion and point == self.exclusion` logic in integer.py?